feat: Added support for Starred operator

nagajaideep opened the pull request:
## Pull Request description
This pull request introduces support for the Starred operator in the ASTX transpiler. The Starred operator is used to represent unpacking operations (e.g., *args) in Python's abstract syntax tree (AST).

-Solve #203 

## How to test these changes

- run `pytest  libs\astx\tests\test_operators.py  -v 
- `pytest libs\astx-transpilers\tests\test_python.py -v  `


## Pull Request checklists

Note:

- [x] Share updated images of the graph representation of the new ASTx node
      proposed in this PR, in both image and ASCII formats. For more
      information, check this Google Colab notebook:
      https://colab.research.google.com/drive/1xXwHmOMkJKFSmhRvn4WYfSAsdDzMnawf?usp=sharing

This PR is a:

- [ ] bug-fix
- [x] new feature
- [ ] maintenance

About this PR:

- [x] it includes tests.
- [x] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:

- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more
      complexity.
- [ ] New and old tests passed locally.

## Additional information

<!-- Add any screenshot that helps to show the changes proposed -->
![image](https://github.com/user-attachments/assets/2f105138-076c-4414-84a1-90e69726c16b)
![image](https://github.com/user-attachments/assets/9c2df678-b22a-4bd0-8f30-d9c928a097af)


<!-- Add any other extra information that would help to understand the changes proposed by this PR -->

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

xmnlab commented on libs/astx/src/astx/operators.py:
```diff
@@ -226,3 +226,31 @@ def get_struct(self, simplified: bool = False) -> ReprStruct:
+
+    def __init__(
+        self,
+        value: DataType,
```
in theory datatype would be ok .. but I am not sure if it would be enough, otherwise maybe it will need to be **Expr**.

could you check the following use cases please:

```
>>> ast.dump(ast.parse("*myfunc()"))
"Module(body=[Expr(value=Starred(value=Call(func=Name(id='myfunc', ctx=Load()), args=[], keywords=[]), ctx=Load()))], type_ignores=[])"
>>> ast.dump(ast.parse("*[1,2,3]"))
'Module(body=[Expr(value=Starred(value=List(elts=[Constant(value=1), Constant(value=2), Constant(value=3)], ctx=Load()), ctx=Load()))], type_ignores=[])'
>>> ast.dump(ast.parse("*a"))
"Module(body=[Expr(value=Starred(value=Name(id='a', ctx=Load()), ctx=Load()))], type_ignores=[])"
>>> ast.dump(ast.parse("*[x for x in range(10)]"))
"Module(body=[Expr(value=Starred(value=ListComp(elt=Name(id='x', ctx=Load()), generators=[comprehension(target=Name(id='x', ctx=Store()), iter=Call(func=Name(id='range', ctx=Load()), args=[Constant(value=10)], keywords=[]), ifs=[], is_async=0)]), ctx=Load()))], type_ignores=[])"
```

in this example we have: Identifier, LiteralList, and ListComprehension

nagajaideep commented:
hey @xmnlab
i have made the requested changes as suggested ,please let know if any further changes are required

nagajaideep commented:
[DO NOT MERGE ]
I NEED TO REMOVE THE ELLIPSE PART

nagajaideep commented:
Hey @xmnlab,  
I've completed the changes as discussed — the code is now working correctly. I’ve updated the pull request and removed the Ellipsis related code from this pull request .

Please have a review when possible and let me know if there are any other suggestions or improvements you'd like to see!

xmnlab commented:
@nagajaideep could you rebase this branch on top of the upstream main branch, please?
